### PR TITLE
[Tooltip] Add support for a hover delay before tooltip is displayed

### PR DESCRIPTION
**Pull request by henryyi**
### WHY are these changes introduced?

Resolves: Shopify/core-workflows/issues/609

Items in the side-nav bar may have a tooltip if the label is truncated. As user may be mousing down the navigation it would not be user friendly if the tooltip pops up while moving down the nav.

tooltip-hover-delay

### WHAT is this pull request doing?

A `hoverDelay` prop is added to the `ToolTip` component which allow consumers to specify the required delay in milliseconds. By default there is no delay.

### How to 🎩

An example can be seen in [storybook](https://polaris.ttd.jita-yi.us.spin.dev/?path=/story/all-components-tooltip--with-hover-delay).

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {Page} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      {/* Add the code you want to test in here */}
    </Page>
  );
}
```

</details>

### 🎩 checklist

- [x] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [x] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [x] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

**Review by jeradg — APPROVED** [review]
Code and 🎩 LGTM!

File: polaris-react/src/components/Tooltip/Tooltip.stories.tsx
Comment (kushp) [inline polaris-react/src/components/Tooltip/Tooltip.stories.tsx]: nit: should*

**Review by kushp — APPROVED** [review]
🎩 already done, code LGTM

File: polaris-react/src/components/Tooltip/Tooltip.tsx
Comment (gui-santos) [inline polaris-react/src/components/Tooltip/Tooltip.tsx:159]: I think my only question about the code is if we need to clearTimeout also when the component unmounts, do we?

If I got it right, we are only clearing it in the MouseLeave event

**Review by gui-santos — CHANGES_REQUESTED** [review]
It looks great!

I got only one question before my approval :)

File: polaris-react/src/components/Tooltip/Tooltip.tsx
Comment (henryyi) [reply polaris-react/src/components/Tooltip/Tooltip.tsx:159]: That's a good point I'll make sure to clear it.

Comment (henryyi) [reply polaris-react/src/components/Tooltip/Tooltip.tsx:159]: I've made the change.

**Review by gui-santos — APPROVED** [review]
Looks great! 🍰

**Review by nabihaalikhan — APPROVED** [review]
looks great 🚢

**Pull request merged**
